### PR TITLE
Access request properties

### PR DESCRIPTION
**Pull request by AlexeySoshin**
Tackles PR #40 
Currently I'm putting those on the context, to have an easy access to them. 

Need to understand what to do with nested objects.
Probably returning a map of maps or some dot notation, eg `actor.name`, would make more sense.

### CODE REVIEW

File: kgraphql-ktor/src/test/kotlin/com/apurebase/kgraphql/FieldsTest.kt [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package com.apurebase.kgraphql
+
+import io.ktor.util.KtorExperimentalAPI
+import kotlinx.serialization.UnstableDefault
+import org.amshove.kluent.shouldBeEqualTo
+import org.junit.jupiter.api.Test
+
+class FieldsTest {
```
Comment (AlexeySoshin) [inline kgraphql-ktor/src/test/kotlin/com/apurebase/kgraphql/FieldsTest.kt:8]: Maybe this should be part of `Schema.execute` tests

File: kgraphql-ktor/src/test/kotlin/com/apurebase/kgraphql/FieldsTest.kt [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    data class Fields(val stuff: String, val moreStuff: String)
+
+    @UnstableDefault
+    @KtorExperimentalAPI
+    @Test
+    fun `All fields can be read is case of a single entity`() {
+        val server = withServer {
+            query("actor") {
+                resolver { ctx: Context ->
+                    val fields = ctx.fields().joinToString(",")
```
Comment (AlexeySoshin) [inline kgraphql-ktor/src/test/kotlin/com/apurebase/kgraphql/FieldsTest.kt:18]: Probably there's a better way to access this that to echoing them in the resolver

File: kgraphql/src/main/kotlin/com/apurebase/kgraphql/schema/model/ast/DocumentNode.kt [hunk 3]
```diff
@@ -1,7 +1,47 @@
...
+                    d
+                else -> null
+            }
+        }
+        defs.forEach { d ->
+            q.add(d.selectionSet)
+        }
+
+        val fields = mutableListOf<NameNode>()
+        while (!q.isEmpty()) {
```
Comment (AlexeySoshin) [inline kgraphql/src/main/kotlin/com/apurebase/kgraphql/schema/model/ast/DocumentNode.kt:24]: Maybe having that BFS doesn't have sense, and we need to fetch only a single level?

**Comment by AlexeySoshin** [comment]
@jeggy WDYT?

**Comment by jeggy** [comment]
> @jeggy WDYT?

I think maybe instead of us providing a simple string array, we should provide the full object. Not sure if that should be the `DocumentNode` or `ExecutionPlan`. Then we could provide some easy to use extension functions to get a string array like what you have done here on top of that.

But I think how exactly this will be used could be up to the developers using this library.

**Comment by AlexeySoshin** [comment]
I'll take a look into that.

**Comment by AlexeySoshin** [comment]
@jeggy Now I'm putting execution nodes on the context. 
Not sure if this could be done, though:
```
resolver { myArg: Int, ctx: Context, gqlNode: Execution.Node ->
```
And I think we're getting a list of them, and not the root one.

**Comment by jeggy** [comment]
I tried to implement this without changing anything regarding the `Context`.

Please take a look here: https://github.com/aPureBase/KGraphQL/pull/99/files#diff-355fb3d2890c86fe5ee3371d7c32ef48R284-R325

**Comment by AlexeySoshin** [comment]
Yep, that's smarted way of doing this :)

I think I'll close this PR then.

**Pull request closed without merging**